### PR TITLE
fix(frontend): handle user in no organization better

### PR DESCRIPTION
**Pull request by JoshFerge**
- if there is no organization slug, don't try to fetch org details
- add a new method `setNoOrganization` on the organization context manager, which will set loading = false, and allow pages to load

In local testing i've observed i'm able to access all of the user organization settings except for organization tokens without the org context being populated.

This means in theory a user should be able to close their account if no org context is loaded. Still probably need to handle redirects better when a user attempts to log in.

I have no idea what the ramifications are hybrid cloud wise, or if there are other settings besides a user not being in an org that can result in there being no org slug. need to test this further

Related: #inc-679

### CODE REVIEW

File: static/app/views/organizationContext.tsx [hunk 1]
```diff
@@ -105,8 +105,8 @@ export function OrganizationContextProvider({children}: Props) {
     if (organization && organization.slug === orgSlug) {
       return;
     }
-
     if (!orgSlug) {
+      OrganizationStore.setNoOrganization();
```
Comment (evanpurkhiser) [inline static/app/views/organizationContext.tsx:109]: This needs to be in a `useEffect`, otherwise you're creating a side effect during render.

Comment (JoshFerge) [reply static/app/views/organizationContext.tsx:109]: I don't understand this, the function that gets called directly below modifies the OrganizationStore (`fetchOrganizationDetails`)

Comment (evanpurkhiser) [reply static/app/views/organizationContext.tsx:109]: Oh sorry. you're right. I didn't expand the diff enough.

File: static/app/views/organizationContainer.tsx [hunk 2]
```diff
@@ -35,7 +35,14 @@ function OrganizationContainer({children}: Props) {
   // organization in context**.
   //
   // TODO(epurkhiser): This scenario desprately should be improved
```
Comment (evanpurkhiser) [inline static/app/views/organizationContainer.tsx:37]: Can you update the comment to reflect that this is now also used for there being no organizations.

**Review by evanpurkhiser — APPROVED** [review]
Seems pretty reasonable to me

**Comment by JoshFerge** [comment]
should i separate out the python template change? or does that matter here

**Review by dashed — APPROVED** [review]

**Comment by dashed** [comment]
> should i separate out the python template change? or does that matter here

doesnt matter since this a broken user experience anyways.

**Pull request merged**